### PR TITLE
Rename `Knowledge*` to `Persisted*`

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?
As we migrate the workspace we've introduced some new types that live alongside the older definitions (to minimise diffs). For the ontology elements we've used the prefix `Persisted*` (for example `PersistedEntityType`) as these represent _persisted_ records, and therefore have the associated metadata such as `ownedById`, etc.

We had also made parallel types for "knowledge" types such as `Entity`, for these we used the prefix `Knowledge*` to avoid conflicts.

This PR renames `Knowledge*` to `Persisted*` for consistency.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1203089535761796/f) _(internal)_

## 🔍 What does this change?

See commits

## 📜 Does this require a change to the docs?

- I don't believe these are mentioned anywhere. As we restructure the types we should improve documentation though to better explain the temporary state and such.

## ⚠️ Known issues

Loads of types are inconsistent

## 🐾 Next steps

- Rework GraphQL types to cover the subgraphs, etc.

## 🛡 What tests cover this?

- Manual testing of the type-editor route
- The "ontology" tests in `integration`
- The "knowledge" tests in `integration`

## ❓ How to test this?

1.  Checkout the branch / view the deployment
2. Ensure you take down the docker deployment and clear all volumes.
3. Run `external-services up --build` in the repo root
4. Inside `packages/hash/integration` in another terminal run
  a.  `LOG_LEVEL="debug" yarn jest "ontology" --maxWorkers 1`  
  b.  `LOG_LEVEL="debug" yarn jest "knowledge" --maxWorkers 1`
5. Also try `yarn dev` and check that login works, and that the `/type-editor` route is not broken.

## 📹 Demo
image
image

### CODE REVIEW

**Review by TimDiekmann — COMMENTED** [review]
LGTM but I think @benwerner01 or @thehabbos007 should take a look as well 🎉

File: packages/hash/api/src/graphql/resolvers/knowledge/model-mapping.ts
Comment (thehabbos007) [inline packages/hash/api/src/graphql/resolvers/knowledge/model-mapping.ts:52]: I agree with you about the casing consistency. I guess his would be a good place to do the `GQL` => `Gql` rename

**Review by thehabbos007 — DISMISSED** [review]
Looks good to me as well. Just a comment about something that has been pointed out earlier

**Comment by Alfred-Mountfield** [comment]
Have fixed conflicts after waiting for the other 2 PRs to merge in. Had more conflicts than expected 🙃 but I have retested, the integration tests still pass and I've messed about with creating some pages and re-ordering them so it all looks good on my end.

**Review by thehabbos007 — APPROVED** [review]

**Review by thehabbos007 — APPROVED** [review]
:egg:

**Pull request merged**
